### PR TITLE
add electrum

### PR DESCRIPTION
**Pull request by theScrabi**
This should add [electrum](https://electrum.org/#home) bitcoin wallet to flathub. Please be aware that I do not have time to keep on maintaining this, as I need a lot of time for my mainline projects.

However maybe one of the electrum developers like @bauerj is willing to look over this or find someone who is interested on maintaining this further.

I will be willing to fix bugs in this script if it is not yet merchable.

### CODE REVIEW

**Comment by theScrabi** [comment]
Can I look up why the build failed?

**Comment by theScrabi** [comment]
Ups the metafile O.o.
Yes I'll add it.

**Comment by nedrichards** [comment]
@theScrabi if you press 'details' on the some checks were not successful or https://flathub.org/builds

**Comment by bauerj** [comment]
How does this work? Can anyone start the build process?

testbot, build org.electrum.electrum

**Comment by theScrabi** [comment]
bot, build org.spesmilo.Electrum

**Comment by theScrabi** [comment]
Yet another try.

**Comment by bauerj** [comment]
testbot, build org.electrum.electrum

**Comment by bauerj** [comment]
testbot, build org.spesmilo.Electrum

afgkjcf

**Comment by theScrabi** [comment]
Yea whatever. I've tried.

File: org.spesmilo.Electrum.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+ <id>org.electrum.electrum</id>
```
Comment (scx) [inline org.spesmilo.Electrum.appdata.xml]: This id (`org.electrum.electrum`) doesn't match the Flatpak ID (`org.spesmilo.Electrum`).

File: org.spesmilo.Electrum.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
```
Comment (scx) [inline org.spesmilo.Electrum.appdata.xml]: The AppData file lacks some data, such as `content_rating` tag (OARS) or `launchable` tag.
https://github.com/flathub/flathub/wiki/AppData-Guidelines
https://hughsie.github.io/oars/generate.html

File: org.spesmilo.Electrum.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <p>- No single point of failure: The server code is open source, anyone can run a server. Private keys can be exported and imported into other wallets</p>
+  <p>- Firewall friendly: The client does not need to open a port, it simply polls the server for updates</p>
+  <p>- Free software: MIT License. Anyone can audit the code</p>
+  <p>- Written in Python. The code is short, and easy to review</p>
+  <p>- Add-ons: third-party plugins are supported</p>
+  <p>- Support for Bitcoin URIs, signed URIs and Bitcoin aliases.</p>
+ </description>
+ <metadata_license>CC0-1.0</metadata_license>
+ <project_license>MIT</project_license>
+ <url type="homepage">https://electrum.org/</url>
```
Comment (scx) [inline org.spesmilo.Electrum.appdata.xml]: You probably should provide more links, like for the [bugtracker](https://github.com/spesmilo/electrum/issues) or [translations](https://crowdin.com/project/electrum).

File: org.spesmilo.Electrum.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ <screenshots>
+  <screenshot>https://en.bitcoin.it/w/images/en/f/fc/Capture-Electrum.png</screenshot>
+ </screenshots>
+ <categories>
```
Comment (scx) [inline org.spesmilo.Electrum.appdata.xml]: Don’t set categories in the AppData file.

https://github.com/flathub/flathub/wiki/AppData-Guidelines#icons-and-categories

File: org.spesmilo.Electrum.desktop [hunk 5]
```diff
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Comment=Lightweight Bitcoin Client
+Exec=electrum %u"
```
Comment (scx) [inline org.spesmilo.Electrum.desktop]: What is it (`"` - the quotation mark) doing here?

File: org.spesmilo.Electrum.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  "runtime-version": "5.12",
+  "sdk": "org.kde.Sdk",
+  "command": "electrum",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--share=network",
+    "--filesystem=host",
```
Comment (scx) [inline org.spesmilo.Electrum.json]: I am afraid that some Flathub maintainers may not like such high privileges.

File: org.spesmilo.Electrum.json [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            "/include"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "electrum",
+      "buildsystem": "simple",
+      "build-commands": [
+        "tar -xzf Electrum-3.3.2.tar.gz",
```
Comment (scx) [inline org.spesmilo.Electrum.json]: You could use the *archive source* for that.

http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html#idm45850111393920

File: org.spesmilo.Electrum.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        "mv Electrum-3.3.2 /app/electrum",
```
Comment (scx) [inline org.spesmilo.Electrum.json]: You should rather use `pip`.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+Electrum flatpak buildfile
```
Comment (scx) [inline README.md]: This README is outdated (wrong id/filename).
Moreover, Flathub maintainers don't like README files in general.

File: org.spesmilo.Electrum.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+ <id>org.electrum.electrum</id>
+ <name>Electrum</name>
+ <developer_name>spesmilo</developer_name>
+ <summary>Lightweight Bitcoin client</summary>
+ <description>
+  <p>Main features are:</p>
+  <p>- Encrypted wallet: the file that contains bitcoin private keys is protected with a password, and never leaves the user's computer</p>
```
Comment (scx) [inline org.spesmilo.Electrum.appdata.xml]: For lists, use the `<ul>` and `<li>` tags instead of `<p>`.

File: org.spesmilo.Electrum.json (hunk 8 above)
Comment (theScrabi) [reply org.spesmilo.Electrum.json]: I've tried pip did not work, as it told me it could not get the app.

File: org.spesmilo.Electrum.json (hunk 6 above)
Comment (theScrabi) [reply org.spesmilo.Electrum.json]: what do you mean do you refere to filesystem=host?

File: org.spesmilo.Electrum.appdata.xml (hunk 3 above)
Comment (theScrabi) [reply org.spesmilo.Electrum.appdata.xml]: I tried to add more url tags but appstream told me they where invalid. I could not find a documentation about how to handle the url tags.

**Comment by theScrabi** [comment]
Yet another try based on @scx suggestions.

**Comment by scx** [comment]
bot, build org.spesmilo.Electrum

**Comment by theScrabi** [comment]
serveimage

File: org.spesmilo.Electrum.appdata.xml (hunk 3 above)
Comment (AsciiWolf) [reply org.spesmilo.Electrum.appdata.xml]: @theScrabi See for example [this](https://gitlab.gnome.org/GNOME/gnome-boxes/blob/master/data/org.gnome.Boxes.appdata.xml.in#L100) from the GNOME Boxes AppData file. ;-)

**Comment by AsciiWolf** [comment]
Any news?

**Comment by theScrabi** [comment]
No time sorry. Vlt domeone else wants to pick this up.

**Comment by A6GibKm** [comment]
I want to pick this up, progress at https://github.com/A6GibKm/flathub/tree/org.electrum.electrum. For some reason the sandbox has access to the entire disk.

**Pull request closed without merging**